omni-reader/README.md

```diff
@@ -0,0 +1,224 @@
+
+# Run comparison with multiple specific models in parallel
+python run_compare_ocr.py --image assets/your_image.jpg --model "gemma3:12b,llama3.2-vision:11b,moondream"
+
+# Run comparison with all available models in parallel
+python run_compare_ocr.py --image assets/your_image.jpg --model all
+```
+
+### Using the Streamlit App
+
+For interactive use, the project includes a Streamlit app:
+
+```bash
+streamlit run app.py
+```
+
+<div align="center">
+  <img src="assets/streamlit.png" alt="Streamlit UI Interface" width="800"/>
+  <p><em>Interactive Streamlit interface for easy document processing and model comparison</em></p>
+</div>
+
+### ### Remote Artifact Storage and Execution
+
+This project supports storing artifacts remotely and executing pipelines on cloud infrastructure. Follow these steps to configure your environment for remote operation:
+
+#### Setting Up Cloud Provider Integrations
+
+Install the appropriate ZenML integrations for your preferred cloud provider:
+
+```bash
+# For AWS
+zenml integration install aws s3 -y
+
+# For Azure
+zenml integration install azure azure-blob -y
+
+# For Google Cloud
+zenml integration install gcp gcs -y
+```
+
+For detailed configuration options and other components, refer to the ZenML documentation:
+
+- [Artifact Stores](https://docs.zenml.io/stacks/artifact-stores)
+- [Orchestrators](https://docs.zenml.io/stacks/orchestrators)
+
+## 📋 Pipeline Architecture
+
+The OCR comparison pipeline consists of the following components:
+
+### Steps
+
+1. **Multi-Model OCR Step**: Processes images with multiple models in parallel
+   - Supports any number of models defined in configuration
+   - Models run in parallel using ThreadPoolExecutor
+   - Each model processes its assigned images with parallelized execution
+   - Progress tracking during batch processing
+2. **Ground Truth Step**: Optional step that uses a reference model for evaluation (default: GPT-4o Mini)
+3. **Evaluation Step**: Compares results and calculates metrics
+
+The pipeline supports configurable models, allowing you to easily swap out the models used for OCR comparison and ground truth generation via the YAML configuration file. It also supports processing any number of models in parallel for more comprehensive comparisons.
+
+### Configuration Management
+
+The new configuration system provides:
+
+- Structured YAML files for experiment parameters
+- Parameter validation and intelligent defaults
+- Easy sharing and version control of experiment settings
+- Configuration generator for quickly creating new experiment setups
+- Support for multi-model configuration via arrays
+- Flexible model selection and comparison
+
+### Metadata Tracking
+
+ZenML's metadata tracking is used throughout the pipeline:
+
+- Processing times and performance metrics
+- Extracted text length and entity counts
+- Comparison metrics between models (CER, WER)
+- Progress tracking for batch operations
+- Parallel processing statistics

```

Comment:
Somehow these things lack a litle bit meaning to me without screenshots or something.. otherwise its all a bit like "meh"